### PR TITLE
API: add git_hash in pd-member's api

### PR DESCRIPTION
**Pull request by mapleFU**
### What problem does this PR solve? 

TiDB-Dashboard require git-hash for components, and previous pd don't have logic about git-hash. So This pr change it and add git hash. 

### What is changed and how it works?

This pr: https://github.com/pingcap/kvproto/pull/599 modify kvproto and add `git_hash` field. This pr add the logic in pd api.

### Release note 

Please put it in 4.0.

### Check List 

Tests 

 - No code

Code changes

 - Has HTTP API interfaces change (Don't forget to [add the declarative for API](https://github.com/pingcap/pd/blob/master/docs/development.md#updating-api-documentation))

Side effects

Related changes

- Need to cherry-pick to the release branch

### CODE REVIEW

**Comment by breeswish** [comment]
@rleungx PTAL, thanks!

**Comment by rleungx** [comment]
@mapleFU CI failed.

File: server/member/leader.go [hunk 1]
```diff
@@ -399,6 +416,20 @@ func (m *Member) SetMemberBinaryVersion(id uint64, releaseVersion string) error
 	return nil
 }
 
+// SetMemberGitHash saves a member's git hash.
+func (m *Member) SetMemberGitHash(id uint64, gitHash string) error {
+	key := m.getMemberGitHashPath(id)
+	txn := kv.NewSlowLogTxn(m.client)
+	res, err := txn.Then(clientv3.OpPut(key, gitHash)).Commit()
```
Comment (nolouch) [inline server/member/leader.go:423]: Why do we need persistence, I understand the version has been saved in the binary file?

Comment (mapleFU) [reply server/member/leader.go:423]: The version is not equal to version, and here we need git_hash. Is there any way to not save git_hash and allowing us to get it?

**Pull request merged**